### PR TITLE
Guac 43 create database tables

### PR DESCRIPTION
**Pull request by ArjunS132**
Created database tables
Created APIs
Created Unit Tests for updated CRUD and APIs

### CODE REVIEW

File: application/backend/app/core/db.py
Comment (mosguinz) [inline application/backend/app/core/db.py:1]: Good work!

File: application/backend/app/models.py [hunk 1]
```diff
@@ -112,3 +119,96 @@ class TokenPayload(SQLModel):
...
+    # Relationships
+    user_facilities: "UserFacility" = Relationship(
+        back_populates="facility_type", cascade_delete=True
+    )
+
+
+class UserFacility(SQLModel, table=True):
+    id: int | None = Field(default=None, primary_key=True)
+    user_id: uuid.UUID = Field(foreign_key="user.id", ondelete="CASCADE")
+    item_type_id: int = Field(foreign_key="facilitytype.id", ondelete="CASCADE")
```
Comment (mosguinz) [inline application/backend/app/models.py:174]: Should the field be called `facility_type`?

File: application/backend/app/tests/api/routes/test_resources.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+from fastapi.testclient import TestClient
+from sqlmodel import Session
+
+from app import crud
+from app.core.config import settings
+from app.models import UserCreate
+from app.tests.utils.utils import random_email, random_lower_string
+
+
+def test_set_resources(client: TestClient, db: Session) -> None:
```
Comment (mosguinz) [inline application/backend/app/tests/api/routes/test_resources.py:10]: Good test case!

**Review by mosguinz — CHANGES_REQUESTED** [review]
Update fields referring to `facility`. In the future, we may choose to nuke the existing `item` table that came with the template.

**Review by mosguinz — APPROVED** [review]

**Review by rizzabears — APPROVED** [review]

**Pull request merged**